libycp/src/YCPBuiltinString.cc

```diff
@@ -299,28 +299,28 @@ s_substring1 (const YCPString &s, const YCPInteger &i1)
     if (s->isAscii())
     {
 	string ss = s->value ();
-	string::size_type start = i1->value();
+	long long start = i1->value();
 
-	if (start < 0 || start > ss.size())
+	if (start < 0 || start > (long long)ss.size())
```

Comment:
ok, as YCP integer is quite limited I think it should not cause problem with string with size between signed and unsigned.